internal/config/types.go

```diff
@@ -0,0 +1,123 @@
+package config
+
+import (
+	"fmt"
+	"sync"
+)
+
+type distributionName = string
+type claimName = string
+type distributionsMap map[distributionName]*Distribution
+type entitlementsMap map[claimName][]distributionName
+
+type Distribution struct {
+	ID     string `json:"id"`
+	Prefix string `json:"prefix"`
+}
+
+// StringPropertiesHash concatenates all string properties in Distribution
+// to form a unique hash
+func (d *Distribution) stringPropertiesHash() string {
+	return fmt.Sprintf("%s%s", d.ID, d.Prefix)
+}
+
+type distributions struct {
+	mu      sync.RWMutex
+	entries distributionsMap
+}
+
+func (d *distributions) Get(key string) (*Distribution, bool) {
```

Comment:
if applying the suggested change below, this would return nil on *Distribution, but if using bool it would be similar to map ok pattern, what do you think is better UX

Reply:
This is an internal API,  In either case a pointer is being returned and a nil check has to be performed.  

It doesn't look like there is a reason to have branching logic in the case `ok` is true and Distribution is nil.  So it seems like returning less information and relying on the nil is enough.

Reply:
makes sense